src/synthcity/plugins/generic/plugin_ddpm.py

```diff
@@ -0,0 +1,193 @@
+"""
+Reference: Kotelnikov, Akim et al. “TabDDPM: Modelling Tabular Data with Diffusion Models.” ArXiv abs/2209.15421 (2022): n. pag.
+"""
+# mypy: disable-error-code=override
```

Comment:
Please re-enable the checks here as well. Both mypy and flake8

The mypy output gives you an indication why the tests are failing: the `_fit` method has a different signature than the Plugin interface. you need to stick to `def _fit(self, X: DataLoader, *args: Any, **kwargs: Any)`. This is why the tests are failing.

Reply:
Sorry I didn't notice that. I have changed the signatures.

Reply:
I had a look why this still fails.

It seems like for Python3.7 or 3.8 you cannot use subscripts `Iterator`. See the diff below.

After that, the tests should run fine, and I will do an in-depth review 

```diff
diff --git a/src/synthcity/plugins/core/models/tabular_ddpm/__init__.py b/src/synthcity/plugins/core/models/tabular_ddpm/__init__.py
index 0618bec..3591000 100644
--- a/src/synthcity/plugins/core/models/tabular_ddpm/__init__.py
+++ b/src/synthcity/plugins/core/models/tabular_ddpm/__init__.py
@@ -55,8 +55,8 @@ class TabDDPM(nn.Module):
 
     def _update_ema(
         self,
-        target_params: Iterator[nn.Parameter],
-        source_params: Iterator[nn.Parameter],
+        target_params: Iterator,
+        source_params: Iterator,
         rate: float = 0.999,
     ) -> None:
         """
         ```